src/UI/Implementation/Component/Popover/ReplaceContentSignal.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace ILIAS\UI\Implementation\Component\Popover;
+
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+use ILIAS\UI\Implementation\Component\Signal;
+
+/**
+ * Class ReplaceContentSignal
+ * @author Stefan Wanzenried <sw@studer-raimann.ch>
+ * @package ILIAS\UI\Implementation\Component\Popover
+ */
+class ReplaceContentSignal extends Signal implements \ILIAS\UI\Component\Popover\ReplaceContentSignal {
+
+	use ComponentHelper;
+
+	/**
+	 * @inheritdoc
+	 */
+	public function withAsyncRenderUrl($url) {
```

Comment:
I could really use a docu how this whole signal thing works. This will also help our futures selfs and new dev. I don't understand how this maintains the connection to the component if this is immutable currently, although I might remember that we talked about it a lot.